656 optimize dockerfile and automate publishing to dockerhub

theboyofdream opened the pull request:
## Description
Issue: https://github.com/TeamShiksha/openlogo/issues/656
This PR optimizes the existing openlogo-express `package/app`  Dockerfile. Reducing docker-image size from **478MB** to **292MB**.

Explaination of `RUN` cmd in docker file:
```Dockerfile
RUN corepack enable && \
    corepack prepare pnpm@latest --activate && \
    pnpm install --prod && \
    rm -rf ~/.pnpm-store /root/.cache /root/.local
```
| CMD																      | Usage																				        |
| --------------------------------------------------- | ------------------------------------------------------------------ |
| `corepack enable`                                           	  | Enables Node's built-in Corepack tool.                                   |
| `corepack prepare pnpm@latest --activate`  	  | Downloads and activates the latest version of `pnpm`.           |
| `pnpm install --prod`                                           | Installs only `dependencies` from `package.json`.                    |
| `rm -rf ~/.pnpm-store /root/.cache /root/.local` | Deletes local caches, pnpm stores/cache & .local folder to reduce final image size. |


## What type of PR is this? (Check all applicable)

- [x] 🔥 Performance Improvements
- [x] 🛠️ CI/CD

## Screenshots (if applicable)

Old docker-image size of openlogo-express is **478MB**.

![openlogo-express-old](https://github.com/user-attachments/assets/0853c413-4444-454b-b7c8-5c8892a02efe)

New  docker-image size of openlogo-express is **292MB**

![openlogo-express-new](https://github.com/user-attachments/assets/39f91a06-d0ac-4b75-8174-56b273acb07f)

## Checklist
- [x] I have performed a self-review of my code  
- [x] I have commented my code, particularly in hard-to-understand areas 
- [x] New and existing unit tests pass locally with my changes

## Extra
- (ui): In `AuthContext.jsx` added dependency `makeRequest` in useEffect because getting lint warning while running `pnpm lint` cmd.

![image](https://github.com/user-attachments/assets/f9922a4d-46e6-4554-b6cc-4f0002a4fc2f)

theboyofdream committed:
updated dockerfile of `package/app` to build optimized image than previous one.
updated `.dockerignore` file to ignore test folder, config file.

fix(ui):added missing dependency in `AuthContext.jsx` file at line 29 to resolve lint error

theboyofdream committed:
Merge branch 'stage' of https://github.com/TeamShiksha/openlogo into 656-optimize-dockerfile-and-automate-publishing-to-dockerhub

theboyofdream committed:
Merge branch 'stage' of https://github.com/TeamShiksha/openlogo into 656-optimize-dockerfile-and-automate-publishing-to-dockerhub

theboyofdream committed:
chore(app): updated Dockerfile and .dockerignore

theboyofdream committed:
empty commit

aps08 commented on packages/app/Dockerfile:
```diff
@@ -1,15 +1,17 @@
 COPY package.json ./
 
-RUN pnpm install
+RUN corepack enable && \
```
explain this statement in the PR description

theboyofdream replied:
yes

aps08 commented on packages/ui/src/contexts/AuthContext.jsx:
```diff
@@ -26,7 +26,7 @@ export const AuthProvider = ({ children }) => {
     const success = await makeRequest();
     if (success) setIsAuthenticated(false);
-  }, [setIsAuthenticated]);
+  }, [setIsAuthenticated, makeRequest]);
```
what is this change about ?

theboyofdream replied:
getting warning while running `pnpm lint`  

![image](https://github.com/user-attachments/assets/189a92f1-6e65-4bcd-b998-094fd3f322f3)

theboyofdream committed:
Merge branch 'stage' of https://github.com/TeamShiksha/openlogo into 656-optimize-dockerfile-and-automate-publishing-to-dockerhub

theboyofdream committed:
removed `pnpm store prune` from Dockerfile because we are deleting pnpm-store in next-Lines.
`.pnpm-store` folder is like a global cache for pnpm
`pnpm store prune` is used to remove package from `.pnpm-store` folder that are not used

theboyofdream commented on packages/app/utils/generatePassword.js:
ran `pnpm format` because getting prettier error while commiting changes
